apps/website/src/components/OpenInCodePen.astro

```diff
@@ -0,0 +1,120 @@
+---
+import outdent from 'outdent';
+import VisuallyHidden from './utils/VisuallyHidden.astro';
+
+type Props = { code: string } & astroHTML.JSX.HTMLAttributes;
+const { code = 'export default () => "hello"', ...rest } = Astro.props;
+
+const css = outdent`
+  @layer reset, itwinui;
+
+  .app-wrapper {
+    height: 100dvh;
+    padding: 1rem;
+  }
+
+  @layer reset {
+    * {
+      box-sizing: border-box;
+    }
+    body {
+      margin: 0;
+    }
+  }
+`;
+const js = outdent`
+  ${code.replace('export default () =>', 'const App = () =>')}
+
+  // -------------------------------------------------------------
+
+  import { createRoot } from 'react-dom/client';
+  import { ThemeProvider } from '@itwin/itwinui-react';
+
+  createRoot(document.body).render(
+    <ThemeProvider className='app-wrapper'>
+      <App />
+    </ThemeProvider>
+  );
+`
+  .replaceAll(`@itwin/itwinui-react`, `https://esm.sh/@itwin/itwinui-react@3`)
+  .replaceAll(`@itwin/itwinui-icons-react`, `https://esm.sh/@itwin/itwinui-icons-react@3`)
+  .replaceAll(`'react'`, `'https://esm.sh/react@18'`)
+  .replaceAll(`react-dom`, `https://esm.sh/react-dom@18`);
```

Comment:
nit: For consistency and robustness, you can use wrap `react-dom` in quotes, similar to how you did for `react`.


```suggestion
  .replaceAll(`'react-dom'`, `'https://esm.sh/react-dom@18'`);
```

Reply:
I did it this way on purpose, to account for `react-dom/client`.

I also used `'react'` (vs `react`) on purpose, because `react` is a more common word that could be used elsewhere.